client/core/core.go

```diff
@@ -1022,6 +1025,13 @@ func (c *Core) handleOrderBookMsg(dc *dexConnection, msg *msgjson.Message) error
 	dc.books[snapshot.MarketID] = ob
 	dc.booksMtx.Unlock()
 
+	// Create the associated epoch queue of the order book.
+	eq := book.NewEpochQueue()
+
+	dc.epochsMtx.Lock()
+	dc.epochs[snapshot.MarketID] = eq
+	dc.epochsMtx.Unlock()
+
```

Comment:
Sorry if I caused confusing in chat, but `book_order` messages, not `orderbook` messages indicate an epoch has closed and is being processed.  An `orderbook` is the entire order book, while `book_order` is an update.
But still as I mentioned in chat, there may not be any `book_order` messages if the closed epoch was empty.  I think the way to detect the epoch close and start of a new one is either a `match_proof`, or a `book_order` with an new epoch index.

Reply:
@chappjc `...or a book_order with an new epoch index.` if I understand correctly that would mean a `BookOrderNote` would have an epoch reference. It currently doesn't, is that an omission?

Reply:
That's true, `book_order` doesn't say which epoch the order was processed in.  But even if it did it did, `book_order` wouldn't be the best indicator since there won't be any for an empty epoch, so I think `match_proof` it the signal to look for that an epoch closed (and was shuffled).

Reply:
Although the epoch index in `epoch_order` should still be checked, and the possibility for an `epoch_order` with the next epoch index coming before `match_proof` should be allowed.